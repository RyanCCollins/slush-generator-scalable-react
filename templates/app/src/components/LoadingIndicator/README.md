## LoadingIndicator Component
A component that acts as a loading indicator.

### Example

```js
<LoadingIndicator isLoading={true} message="Loading" />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **isLoading**    | Bool   |  True      | Whether the loading indicator is currently loading or not
| **message**    | Bool   | "Loading"  | The message to display with the spinner.


### Other Information
Will be centered in whatever box it is in.
