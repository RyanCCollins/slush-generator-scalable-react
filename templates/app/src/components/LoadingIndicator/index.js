import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Spinning from 'grommet-udacity/components/icons/Spinning';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';

const LoadingIndicator = ({
  isLoading,
  message,
}) => (
  <Box
    align="center"
    justify="center"
    className={styles.loadingIndicator}
  >
    {isLoading &&
      <Box
        align="center"
        justify="center"
      >
        <Spinning />
        <Heading tag="h3" align="center">{message}</Heading>
      </Box>
    }
  </Box>
);

LoadingIndicator.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  message: PropTypes.string.isRequired,
};

LoadingIndicator.defaultProps = {
  isLoading: true,
  message: 'Loading',
};

export default cssModules(LoadingIndicator, styles);
