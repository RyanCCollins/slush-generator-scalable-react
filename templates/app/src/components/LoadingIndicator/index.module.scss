.loadingIndicator {
  margin-top: 20px;
}
