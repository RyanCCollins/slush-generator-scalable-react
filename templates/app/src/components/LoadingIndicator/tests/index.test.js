import LoadingIndicator from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<LoadingIndicator />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <LoadingIndicator isLoading />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
