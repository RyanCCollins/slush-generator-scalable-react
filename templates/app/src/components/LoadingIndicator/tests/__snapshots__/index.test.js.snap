exports[`<LoadingIndicator /> should render with default props 1`] = `
<Box
  align="center"
  announce={false}
  className={undefined}
  direction="column"
  focusable={true}
  justify="center"
  pad="none"
  responsive={true}
  tag="div">
  <Box
    align="center"
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="none"
    responsive={true}
    tag="div">
    <Spinning />
    <Heading
      align="center"
      tag="h3">
      Loading
    </Heading>
  </Box>
</Box>
`;
